workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QualifiedLeadsTemplateValidatorController.java

```diff
@@ -0,0 +1,173 @@
+import org.apache.poi.ss.usermodel.CellType;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.ss.usermodel.WorkbookFactory;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+/**
+ * @author Eduardo Diniz
+ */
+@RequestMapping("/file-validation")
+@RestController
+public class QualifiedLeadsTemplateValidatorController
+	extends BaseRestController {
+
+	@PostMapping("/validate")
+	public ResponseEntity<String> validateFile(
+		@AuthenticationPrincipal Jwt jwt,
+		@RequestParam("file") MultipartFile file,
+		HttpServletRequest httpServletRequest) {
+
+		try {
+			Set<String> fileColumns = _extractColumns(file);
+			Set<String> templateColumns = _extractColumnsFromTemplate();
+
+			if (fileColumns.equals(templateColumns)) {
+				return ResponseEntity.ok("File is valid.");
+			}
+
+			return ResponseEntity.badRequest(
+			).body(
+				"File is invalid. Columns do not match the template."
+			);
+		}
+		catch (IOException ioException) {
+			return ResponseEntity.status(
+				500
+			).body(
+				"Error processing file: " + ioException.getMessage()
+			);
+		}
+	}
+
+	private Set<String> _extractColumns(MultipartFile file) throws IOException {
+		String contentType = file.getContentType();
+
+		if ((contentType != null) && contentType.equals("text/csv")) {
+			return _extractColumnsFromCSV(file.getInputStream());
+		}
+
+		return _extractColumnsFromExcel(file.getInputStream());
+	}
+
+	private Set<String> _extractColumnsFromCSV(InputStream inputStream)
+		throws IOException {
+
+		Set<String> columns = new LinkedHashSet<>();
+
+		try (BufferedReader reader = new BufferedReader(
+				new InputStreamReader(inputStream))) {
+
+			String line;
+			int lineNumber = 0;
+
+			while (((line = reader.readLine()) != null) && (lineNumber < 2)) {
+				String[] headers = line.split(",");
+
+				for (String header : headers) {
+					String normalizedHeader = _normalizeHeader(header);
+
+					if (!normalizedHeader.isEmpty()) {
+						columns.add(normalizedHeader);
+					}
+				}
+
+				lineNumber++;
+			}
+		}
+
+		return columns;
+	}
+
+	private Set<String> _extractColumnsFromExcel(InputStream inputStream)
+		throws IOException {
+
+		Set<String> columns = new LinkedHashSet<>();
+
+		try (Workbook workbook = WorkbookFactory.create(inputStream)) {
+			Sheet sheet = workbook.getSheetAt(0);
+
+			for (int rowIndex = 0; rowIndex <= 1; rowIndex++) {
```

Comment:
can we switch this to an enhanced for loop and clean up to be a bit more readable? 

im thinking something like this, what do you think?
```
for (Row row : sheet) {
    if (row == null || (row.getRowNum() > 1)) {
        break;
    }

    for (Cell cell : row) {
        String cellValue = _normalizeHeader(_getCellValueAsString(cell));

        if (!cellValue.isEmpty()) {
            columns.add(cellValue);
        }
    }
}
```

Reply:
Nice suggestion :rocket:  
Thanks!